File: Data Structures & Algorithms/longest-substring-without-duplicates/submission-0.cpp
class Solution {
public:
    int lengthOfLongestSubstring(string s) {
        int left=0,right=0;
        unordered_set<int>st;
        int ans  = 0;
        int n = s.size();
        while(left<=n-1 and right<=n-1){
            if(st.find(s[right])==st.end()){
                st.insert(s[right]);
                right++;
                ans = max(ans,int(st.size()));
                
            }
            else {
                ans = max(ans,int(st.size()));
                st.erase(s[left]);
                left++;
                
            }
        }
        return ans;
    }
};

File: Data Structures & Algorithms/balanced-binary-tree/submission-0.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right

class Solution:
    def isBalanced(self, root: Optional[TreeNode]) -> bool:
        self.bal = True
        def height(root):
            if not root:
                return 0

            a = height(root.left)
            b=height(root.right)
            if abs(a-b)>1:
                self.bal=False
            return 1+max(a,b)
        height(root)
        return self.bal
